solr/solrj/src/java/org/apache/solr/common/assign/ReplicaAssignStrategy.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.common.assign;
+
+
+import org.apache.solr.client.solrj.cloud.SolrCloudManager;
+import org.apache.solr.client.solrj.cloud.autoscaling.Suggester;
+import org.apache.solr.common.MapWriter;
+import org.apache.solr.common.cloud.Replica;
+import org.apache.solr.common.params.CollectionParams;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The implementation class can be stored in clusterprops.json as follows
+ * {
+ *     assign-strategy : {
```

Comment:
It should be possible to load and use multiple named strategies. I can imagine a common default strategy that uses Policy framework for some collections, and an override in collection config to use another strategy for other collections where Policy doesn't work well.

Reply:
I would strongly recommend against that. A violation for one strategy may not be same for another